### PR TITLE
tarfs: Symlink changes

### PR DESCRIPTION
**Pull request by crozzy**
Make files with symlinks in their paths accessible both through the symlink path and the original path. This also takes into account when tars have been concatenated and symlinked dirs have been previously defined as normal dirs.

Signed-off-by: crozzy <joseph.crosland@gmail.com>

### CODE REVIEW

**Review by hdonnay — APPROVED** [review]
SGTM, but I rewrote large parts of it

File: pkg/tarfs/tarfs_test.go [hunk 1]
```diff
@@ -379,3 +379,88 @@ func TestKnownLayers(t *testing.T) {
...
+			if err != nil {
+				return err
+			}
+			t.Logf("path: %q", path)
+			seen, ok := test.expectedFS[path]
+			if !ok {
+				t.Fatalf("didn't expect path %s", path)
+			}
+			if seen {
+				t.Logf("we already saw this path: %s", path)
```
Comment (crozzy) [inline pkg/tarfs/tarfs_test.go:452]: I think, as the "exactly once" problem is solved with your update, this should be now be fatal (so we can catch any regressions).

**Review by crozzy — COMMENTED** [review]
Also LGTM, just a comment

File: pkg/tarfs/tarfs_test.go (hunk 1 above)
Comment (crozzy) [reply pkg/tarfs/tarfs_test.go:452]: I should just push this change

**Review by hdonnay — APPROVED** [review]

**Pull request merged**